test/notify/send-email.test.js

```diff
@@ -0,0 +1,164 @@
+const flatten = require('flat')
+const env = require('../env')
+const mockContext = require('../mock-context')
+const mockMessage = require('../mock-context')
+const mockReference = require('../mock-reference')
+
+let NotifyClient
+let validateEmail
+let sendEmail
+let emailAddress
+let defaultEmailAddress
+let defaultReference
+
+describe('send email', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    jest.resetModules()
+
+    jest.mock('../../ffc-pay-alerts/notify/validate-email')
+    validateEmail = require('../../ffc-pay-alerts/notify/validate-email')
+
+    NotifyClient = require('notifications-node-client').NotifyClient
+    jest.mock('notifications-node-client')
+
+    sendEmail = require('../../ffc-pay-alerts/notify/send-email')
+
+    emailAddress = 'test@test.com'
+
+    defaultEmailAddress = env.notifyEmailAddress
+    defaultReference = ''
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('should correctly configure Notify client on module import', async () => {
+    expect(NotifyClient).toHaveBeenCalledTimes(1)
+    expect(NotifyClient).toHaveBeenCalledWith(env.notifyApiKey)
+  })
+
+  test('should call validateEmail when a valid message is received', async () => {
+    await sendEmail(mockContext, mockMessage)
+
+    expect(validateEmail).toHaveBeenCalled()
+  })
+
+  test('should call validateEmail with default emailAddress when sendEmail is called without a given emailAddress', async () => {
+    await sendEmail(mockContext, mockMessage)
+
+    expect(validateEmail).toHaveBeenCalledWith(mockContext, defaultEmailAddress)
+  })
+
+  test('should not reject on validateEmail with default emailAddress when sendEmail is called without a given emailAddress', async () => {
+    await sendEmail(mockContext, mockMessage)
+
+    expect(validateEmail).not.toThrow()
+  })
+
+  test('should not reject with default emailAddress when sendEmail is called without a given emailAddress', async () => {
+    const wrapper = async () => {
+      await sendEmail(mockContext, mockMessage)
+    }
+
+    expect(wrapper).not.toThrow()
+  })
+
+  test('should throw an error when validateEmail rejects', async () => {
+    validateEmail.mockRejectedValue(new Error('must be a string'))
+
+    const wrapper = async () => {
+      await sendEmail(mockContext, mockMessage)
+    }
+
+    expect(wrapper).rejects.toThrow()
+  })
+
+  test('should throw Error when validateEmail rejects', async () => {
+    validateEmail.mockRejectedValue(new Error('must be a string'))
+
+    const wrapper = async () => {
+      await sendEmail(mockContext, mockMessage)
+    }
+
+    expect(wrapper).rejects.toThrow(Error)
+  })
+
+  test('should throw Error when validateEmail rejects', async () => {
```

Comment:
this looks like a duplicate test of the one on line 78

Reply:
Well picked up -- removed